src/firefly/js/charts/ui/PlotlyChartArea.jsx

```diff
@@ -63,7 +65,15 @@ export class PlotlyChartArea extends PureComponent {
             doingResize = true;
         }
         const showlegend = data.length > 1;
-        let pdata = data.map((e) => Object.assign({}, e)); // create shallow copy of data elements to avoid sharing x,y,z arrays
+
+        // put the active trace after all inactive traces
+        let pdata = data.reduce((rdata, e, idx) =>  {
+            (idx !== activeTrace) && rdata.push(e);
+            return rdata;
+        }, []);
+
+        pdata.push(data[activeTrace]);
+        //let pdata = data.map((e) => Object.assign({}, e)); // create shallow copy of data elements to avoid sharing x,y,z arrays
```

Comment:
I think this piece of code should not be commented. It is making sure expanded and unexpanded charts are not sharing the data.

To see the bug, open a heatmap in expanded mode, go to options, change bin size to 10 x 10, apply. The chart is not changing.  The reason is the expanded chart and unexpanded chart underneath share the data, and the change is first applied to unexpanded chart. Then when expanded chart optimize is checking for updates, the data appear updated.

Now the bug is back. I should have added better explanation in the comment.

Reply:
The code is modified to collect the trace data in shallow copy style.